plugin/pkg/shoot/validator/admission.go

```diff
@@ -781,8 +788,16 @@ func (c *validationContext) validateProvider(a admission.Attributes) field.Error
 				detail := fmt.Sprintf("machine type '%s' does not support CPU architecture '%s' or is unavailable in at least one zone, supported machine types are: %+v", worker.Machine.Type, *worker.Machine.Architecture, validMachineTypes)
 				allErrs = append(allErrs, field.Invalid(idxPath.Child("machine", "type"), worker.Machine.Type, detail))
 			}
-			if ok, validMachineImages := validateMachineImagesConstraints(c.cloudProfile.Spec.MachineImages, worker.Machine, oldWorker.Machine); !ok {
-				detail := fmt.Sprintf("machine image '%s' does not support CPU architecture '%s' or is expired, valid machine images are: %+v", worker.Machine.Image, *worker.Machine.Architecture, validMachineImages)
+
+			kubeletVersion, err := helper.CalculateEffectiveKubernetesVersion(controlPlaneVersion, worker.Kubernetes)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(idxPath, worker.Kubernetes, "cannot determine effective Kubernetes version for worker pool"))
```

Comment:
The error can occur only if version is specified.
```suggestion
				allErrs = append(allErrs, field.Invalid(idxPath.Child("version"), worker.Kubernetes.Version, "cannot determine effective Kubernetes version for worker pool"))
```